File: FrontEnd/src/app/games/game.ts
export class Game {
    nombre:string|null;
    plataforma1:string|null;
    plataforma2:string|null;
    plataforma3:string|null;
    plataformas:string|null;
    desarrolladora:string|null;
    genero1:string|null;
    genero2:string|null;
    genero3:string|null;
    generos:string|null;
    notaMedia:number|null;
    fechaLanzamiento:Date|null;
    fechaRegistro:Date|null;


    constructor(){
        this.nombre = null;
        this.plataforma1 = null;
        this.plataforma2 = null;
        this.plataforma3 = null;
        this.plataformas = null;
        this.desarrolladora = null;
        this.genero1 = null;
        this.genero2 = null;
        this.genero3 = null;
        this.generos = null;
        this.notaMedia = null;
        this.fechaLanzamiento = null;
        this.fechaRegistro = null;
    }
}

File: games_service/src/main/java/upm/tfg/b190222/games_service/entity/Game.java
package upm.tfg.b190222.games_service.entity;

import java.time.LocalDate;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
@Table(name="Game")
public class Game {
    
    @Id
    @Column(name="nombre")
    private String nombre;

    @Column(name="plataforma")
    private String plataforma;

    @Column(name="desarrolladora")
    private String desarrolladora;

    @Column(name="genero1")
    private String genero1;

    @Column(name="genero2")
    private String genero2;

    @Column(name="genero3")
    private String genero3;

    @Column(name="notaMedia")
    private float notaMedia;

    @Column(name="fechaLanzamiento")
    private LocalDate fechaLanzamiento;

    @Column(name="fechaRegistro")
    private LocalDate fechaRegistro;
}
